pika/adapters/base_connection.py

```diff
@@ -239,7 +241,10 @@ def _get_error_code(self, error_value):
         return None
 
     def _flush_outbound(self):
-        """Call the state manager who will figure out that we need to write."""
+        """write early, if the socket will take the data why not get it out
+        there asap.
+        """
+        self._handle_write()
```

Comment:
@vitaly-krugl, thanks. Though I think a lot of your gotchas will have to have been handled at the python wrapper layer, the interface exposed to python is reasonably straightforward. We'll see how we go.